src/main/java/vazkii/botania/common/block/subtile/generating/SubTileRafflowsia.java

```diff
@@ -22,20 +23,61 @@
 
 import javax.annotation.Nullable;
 
+import java.util.LinkedList;
+import java.util.List;
+import java.util.ListIterator;
+
 public class SubTileRafflowsia extends TileEntityGeneratingFlower {
-	public static final String TAG_LAST_FLOWER = "lastFlower";
+	public static final String TAG_LAST_FLOWERS = "lastFlowers";
 	public static final String TAG_LAST_FLOWER_TIMES = "lastFlowerTimes";
+	public static final String TAG_STREAK_LENGTH = "streakLength";
 
 	@Nullable
-	private Block lastFlower;
-	private int lastFlowerTimes;
+	private List<String> lastFlowers = new LinkedList<>();
```

Comment:
why strings instead of blocks?

Reply:
Was easier to figure out how to serialize, LOL

Reply:
let's not stringly type, you can recover the Block from the string by doing `ResourceLocation.tryCreate` then `Registry.BLOCK.getOrEmpty` (name may differ)